app/api/common/lib/transactions/ISignRequest.js

```diff
@@ -9,6 +10,6 @@ export interface ISignRequest<T> {
   uniqueSenderAddresses(): Array<string>;
   receivers(includeChange: boolean): Array<string>;
   isEqual(tx: ?mixed): boolean;
-
+  txMetadata(): RustModule.WalletV4.TransactionMetadata | void;
```

Comment:
probably don't need to make this part of the `ISignRequest` interface and instead only make it part of the Shelley sign request. Similar to withdrawals, etc. we can just check the type using `instanceof` then case when needed.